### PR TITLE
646 - Add info on why the TrieStore uses account hashes

### PR DESCRIPTION
**Pull request by ipopescu**
### Related links

- #646 

### Changes

- Updated [design](https://docs.casperlabs.io/design/casper-design/#accounts-head) and [keys](https://docs.casperlabs.io/dapp-dev-guide/keys/) pages as requested in the ticket

### Notes

- Ran the docs site locally
- Since the technical details are clear, I am just adding @ACStoneCL as reviewer

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/keys.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-The Casper blockchain uses an on-chain [account-based model](/design/casper-design.md/#accounts-head), uniquely identified by an `AccountHash` derived from a specific `PublicKey`.
+The Casper blockchain uses an on-chain [account-based model](/design/casper-design.md/#accounts-head), uniquely identified by an `AccountHash` derived from a specific `PublicKey`. The `AccountHash` is a 32-byte hash derived from any of the supported `PublicKey` variants below to standardize keys that can vary in length.
 
 By default, a transactional interaction with the blockchain takes the form of a `Deploy` cryptographically signed by the key-pair corresponding to the `PublicKey` used to create the account.
 
 The Casper platform supports two types of keys for creating accounts and signing transactions: 
-- [ed25519](#eddsa-keys) keys, which use the Edwards-curve Digital Signature Algorithm (EdDSA)
-- [secp256k1](#ethereum-keys) keys, commonly known as Ethereum keys
+- [ed25519](#eddsa-keys) keys, which use the Edwards-curve Digital Signature Algorithm (EdDSA) and are 66-byte long
+- [secp256k1](#ethereum-keys) keys, commonly known as Ethereum keys, which are 68-byte long
```
Comment (ACStoneCL) [inline source/docs/casper/dapp-dev-guide/keys.md:9]: Should we pluralize bytes for these two?

File: source/docs/casper/design/casper-design.md [hunk 2]
```diff
@@ -164,7 +164,13 @@ All these features are accessible via functions in the [Casper External FFI](htt
...
+The Casper blockchain uses an on-chain account-based model, uniquely identified by an `AccountHash` derived from a specific `PublicKey`. The [global state trie store](#global-state-trie) requires all keys to be the same length, so the AccountHash is a 32-byte derivative used to abstract any of the supported public key variants.
+
+The Casper platform supports two types of keys for creating accounts and signing transactions: 
+- [ed25519](/dapp-dev-guide/keys.md#eddsa-keys) keys, which use the Edwards-curve Digital Signature Algorithm (EdDSA) and are 66-byte long
+- [secp256k1](/dapp-dev-guide/keys.md#ethereum-keys) keys, commonly known as Ethereum keys, which are 68-byte long
```
Comment (ACStoneCL) [inline source/docs/casper/design/casper-design.md:171]: Same as above, the 66-byte/68-byte format here sounds weird to me, and we use 66-bytes/68-bytes earlier as well.

**Review by ACStoneCL — APPROVED** [review]
Just a small nit, otherwise LGTM!

File: source/docs/casper/dapp-dev-guide/keys.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/dapp-dev-guide/keys.md]: Yes, good catch, thanks!

File: source/docs/casper/design/casper-design.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/design/casper-design.md]: Updated to use "bytes".

**Comment by ipopescu** [comment]
@ramujin @ShalmaliCL fyi.

**Pull request merged**